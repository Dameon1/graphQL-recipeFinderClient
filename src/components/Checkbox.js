import React from "react";
import { IngredientBox } from "./styles";

export class Checkbox extends React.Component {
  state = {
    isChecked: false
  };

  toggleCheckboxChange = () => {
    const { handleCheckboxChange, label } = this.props;
    this.setState(({ isChecked }) => ({ isChecked: !isChecked }));
    handleCheckboxChange(label);
  };

  render() {
    const { label } = this.props;
    const { isChecked } = this.state;

    return (
      <IngredientBox onClick={this.toggleCheckboxChange}>
        <label>
          <input
            type="checkbox"
            value={label}
            checked={isChecked}
            onChange={this.toggleCheckboxChange}
          />
          {label}
        </label>
      </IngredientBox>
    );
  }
}

export default Checkbox;