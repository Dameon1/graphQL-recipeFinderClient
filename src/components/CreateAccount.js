//TODO: ****REMOVE "SIGNUP_MUTATION TO ITS OWN ACTION*********

import React from "react";
import { Mutation } from "react-apollo";
import gql from "graphql-tag";

import { Form, UserSignIn, SignInButton, SignInInput, Label } from "./styles";

import Error from "./ErrorMessage";
import { navigate } from "@reach/router";
import { CURRENT_USER_QUERY } from "./User";

const SIGNUP_MUTATION = gql`
  mutation SIGNUP_MUTATION($username: String!, $password: String!) {
    createUser(username: $username, password: $password) {
      username
    }
  }
`;

function routeToHome() {
  navigate("/dashboard");
}

export class CreateAccount extends React.Component {
  state = {
    password: "",
    username: ""
  };
  saveToState = e => {
    this.setState({ [e.target.name]: e.target.value });
  };
  render() {
    return (
      <Mutation
        mutation={SIGNUP_MUTATION}
        variables={this.state}
        refetchQueries={[{ query: CURRENT_USER_QUERY }]}
      >
        {(signup, { data, error, loading }) => {
          if (data) {
            routeToHome(data);
          }
          return (
            <UserSignIn>
              <Form
                method="post"
                onSubmit={async e => {
                  e.preventDefault();
                  await signup();
                  this.setState({ password: "", username: "" });
                }}
              >
                <Error error={error} />
                <Label htmlFor="username">Username</Label>
                <SignInInput
                  name="username"
                  placeholder="username"
                  value={this.state.username}
                  onChange={this.saveToState}
                  type="text"
                />
                <Label htmlFor="password">Password</Label>
                <SignInInput
                  type="password"
                  name="password"
                  placeholder="password"
                  value={this.state.password}
                  onChange={this.saveToState}
                />
                <SignInButton
                  type="submit"
                  disabled={this.props.pristine || this.props.submitting}
                >
                  Register
                </SignInButton>
              </Form>
            </UserSignIn>
          );
        }}
      </Mutation>
    );
  }
}

export default CreateAccount;
