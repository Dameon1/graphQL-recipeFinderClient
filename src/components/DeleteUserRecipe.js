import React from "react";
import { Mutation } from "react-apollo";
import PropTypes from "prop-types";
import { DELETE_USER_RECIPE_MUTATION } from '../actions';

const DeleteUserRecipe = props => {
  return (
    <Mutation {...props} mutation={DELETE_USER_RECIPE_MUTATION} variables={props.variables}>
      {payload => props.children(payload)}
    </Mutation>
  );
};

DeleteUserRecipe.propTypes = {
  children: PropTypes.func.isRequired
};

export default DeleteUserRecipe;