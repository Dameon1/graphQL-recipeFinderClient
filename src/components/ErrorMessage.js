import React from "react";
import { ErrorStyles } from "./styles";
import PropTypes from "prop-types";

const ErrorMessage = ({ error }) => {
  if (!error || !error.message) return null;
  if (
    error.networkError &&
    error.networkError.result &&
    error.networkError.result.errors.length
  ) {
    return error.networkError.result.errors.map((error, i) => (
      <ErrorStyles key={i}>
        <p data-test="graphql-error">
          <strong>Shoot!</strong>
          {error}
        </p>
      </ErrorStyles>
    ));
  }
  return (
    <ErrorStyles>
      <p data-test="graphql-error">
        <strong>Shoot!</strong>
        {error.message}
      </p>
    </ErrorStyles>
  );
};

ErrorMessage.defaultProps = {
  error: {}
};

ErrorMessage.propTypes = {
  error: PropTypes.object
};

export default ErrorMessage;
