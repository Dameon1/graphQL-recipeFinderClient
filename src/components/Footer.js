//TODO ** CREATE STYLED COMPONENTS**

import React from "react";
import {FooterContainer,FooterText,FooterLink} from './styles';
import "./styles/footer.css";

export default function Footer() {
  return (
    <FooterContainer>
      <FooterText>
        Powered by{" "}
        <FooterLink
          href="https://spoonacular.com/"
          target="blank"
        >
          @Spoonacular
        </FooterLink>
      </FooterText>
    </FooterContainer>
  );
}