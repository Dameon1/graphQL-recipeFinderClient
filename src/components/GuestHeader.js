import React from "react";
import { Link } from "@reach/router";
import { NavButton } from "./styles";

export default class GuestHeader extends React.Component {
  render() {
    return (
      <nav>
        <Link to="/signIn">
          <NavButton>Login</NavButton>
        </Link>
        <Link to="/signUp">
          <NavButton>Sign up</NavButton>
        </Link>
        <Link to="/dashboard">
          <NavButton>Dashboard</NavButton>
        </Link>
        <Link to="/">
          <NavButton>Home</NavButton>
        </Link>
      </nav>
    );
  }
}
