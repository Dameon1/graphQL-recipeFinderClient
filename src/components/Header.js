import React from "react";
import { Link } from '@reach/router';
import User from './User';
import './styles/header.css';
import Spinner from 'react-spinkit';
import SignOut from './SignOut';
// import { signUserOut } from '../../actions/auth';
// import { signingUserOut } from "../../actions/userActions";

 export default class Header extends React.Component {

//   handleSignOut() {
//    this.props.dispatch(signingUserOut());
//    this.props.dispatch(signUserOut());
//   };

//TODO SEPARATE HEADERS INTO SEPARATE FILES
  render() {
    return (      
      <User>
        {(data) => {
          console.log(data.loading);
          if(data.loading) {
           return <Spinner spinnername="circle" fadeIn='none' />;
          }
          if(!data.data.me) {
            return (
              <div className='header'>
                <h1 className="headerText">
                  what2eat
                </h1>
                <div className="navLinks">
                  <Link to="/signIn" >
                    <button className="navLinkButtons">Login</button>
                  </Link>
                  <Link to="/signUp" >
                    <button className="navLinkButtons">Sign up</button>
                  </Link>  
                  <Link to="/dashboard" >
                    <button className="navLinkButtons" >Dashboard</button>
                  </Link>
                  <Link to="/" >
                    <button className="navLinkButtons">Home</button>
                  </Link>
                </div>
              </div>
            )
          } else {
            return (
              <div className='header'>
                <h1 className="headerText">
                  what2eat
                </h1>
                <div className="navLinks">
                  <Link to="/dashboard" >
                    <button className="navLinkButtons">Dashboard</button>
                  </Link>
                  <Link to="/myRecipes" >
                    <button className="navLinkButtons">My recipes</button>
                  </Link> 
                  <Link to="/" >
                    <SignOut />
                  </Link> 
                </div>
              </div>
          )
          }
        }}
      </User>
    )

    // if (!this.props.loggedIn) {
    //   return (
        
    // )}
     
  };
};
