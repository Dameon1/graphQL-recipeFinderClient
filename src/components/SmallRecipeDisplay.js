
import React from "react";
import { Link } from "@reach/router";
import "./styles/multipleRecipesDisplay.css";
import { MultiRecipeButton, MultiRecipeImage, SmallRecipeContainer,SmallRecipeTitleText } from "./styles";

export class SmallRecipeDisplay extends React.Component {
  render() {
    const { props } = this.props;
    return (
      <SmallRecipeContainer key={props.index} value={props.id}>
        <SmallRecipeTitleText>{props.title}</SmallRecipeTitleText>
        <MultiRecipeImage src={props.image} alt={`Of ${props.title}`} />
        <p>Ingredients used: {props.usedIngredientCount}</p>
        <p>Missing ingredients: {props.missedIngredientCount}</p>
        <Link to={`/singleRecipe/${props.id}`}>
          <MultiRecipeButton>Get Recipe</MultiRecipeButton>
        </Link>
      </SmallRecipeContainer>
    );
  }
}

export default SmallRecipeDisplay;
