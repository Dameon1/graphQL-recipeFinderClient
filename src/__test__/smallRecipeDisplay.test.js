import React from "react";
import { shallow } from "enzyme";
import { SmallRecipeDisplay } from "../components/SmallRecipeDisplay";
import toJSON from "enzyme-to-json";

//TODO: add an actual recipe

describe("<SmallRecipeDisplay />", () => {
  it("Renders without crashing", () => {
    const configs = {
      default: true,
      label: "My Label",
      element: "myElement",
      apiRecipes: [12232, 1221],
      recipes: [12232]
    };
    const props = {
      props: {
      index: 1,
      id:101,
      title:"This is a test title",
      image: "Test url image String",
      usedIngredientCount: 3,
      missedIngredientCount: 2
    }}
    const dispatch = jest.fn();    
    const wrapper = shallow(<SmallRecipeDisplay {...props}/>);
    //console.log(wrapper.debug());
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});