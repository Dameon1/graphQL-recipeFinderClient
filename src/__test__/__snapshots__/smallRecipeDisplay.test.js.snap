// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SmallRecipeDisplay /> Renders without crashing 1`] = `
<styled.div
  key="1"
  value={101}
>
  <styled.h2>
    This is a test title
  </styled.h2>
  <styled.img
    alt="Of This is a test title"
    src="Test url image String"
  />
  <p>
    Ingredients used: 
    3
  </p>
  <p>
    Missing ingredients: 
    2
  </p>
  <ForwardRef
    to="/singleRecipe/101"
  >
    <styled.button>
      Get Recipe
    </styled.button>
  </ForwardRef>
</styled.div>
`;
